jetty-ee10/jetty-ee10-osgi/jetty-ee10-osgi-boot/src/main/java/org/eclipse/jetty/ee10/osgi/boot/OSGiMetaInfConfiguration.java

```diff
@@ -289,24 +279,26 @@ private List<Resource> getBundleAsResource(ResourceFactory resourceFactory, Bund
             {
                 if (FileID.isJavaArchive(f.getName()) && f.isFile())
                 {
-                    resources.add(resourceFactory.newResource(f.toPath()));
+                    // add *.jar as jar files
+                    resources.add(resourceFactory.newJarFileResource(f.toPath().toUri()));
                 }
                 else if (f.isDirectory() && f.getName().equals("lib"))
                 {
-                    for (File f2 : file.listFiles())
+                    for (File libFile : file.listFiles())
                     {
-                        if (FileID.isJavaArchive(f2.getName()) && f2.isFile())
+                        if (FileID.isJavaArchive(libFile.getName()) && libFile.isFile())
                         {
-                            resources.add(resourceFactory.newResource(f.toPath()));
+                            // add lib/*.jar as jar files
+                            resources.add(resourceFactory.newJarFileResource(f.toPath().toUri()));
```

Comment:
Again, too slow.

This is like all of our old code that constantly converted from Strings to File to Strings to URI to Strings to File to URL to Strings to URI.

Keep it how it will be used.